EIPS/eip-5164.md

```diff
@@ -0,0 +1,209 @@
+---
+eip: 5164
+title: Cross-Chain Execution
+description: Defines an interface that supports execution across EVM networks.
+author: Brendan Asselstine (@asselstine), Pierrick Turelier (@PierrickGT), Anna Carroll (@anna-carroll), Hadrien Croubois (@Amxx), Nam Chu Hoai (@nambrot), Georgios (@geogons), Theo Gonella (@mintcloud), Rafael Solari (@rsolari), Auryn Macmillan (@auryn-macmillan), Nathan Ginnever (@nginnever)
+discussions-to: https://ethereum-magicians.org/t/eip-5164-cross-chain-execution/9658
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-06-14
+---
+
+## Abstract
+
+This specification defines a cross-chain execution interface for EVM-based blockchains. Developers will be able to have contracts call contracts on other chains. There are two parts to this spec: the "sending" interface and the "executing" interface. The "sending" interface specifies the interface to send a call to another chain; this interface must be implemented by bridges. The "executing" interface must be implemented by the application; it is an interface for executing cross-chain calls. This specification is agnostic of the bridge implementation; it simply standardizes the call interfaces.
+
+## Motivation
+
+Many Ethereum protocols need to coordinate state changes across multiple EVM-based blockchains. These chains often have native or third-party bridges that allow Ethereum contracts to execute code. However, bridges have different APIs so bridge integrations are bespoke.
+
+Bridge technology is improving rapidly, and there are multiple bridges to choose from. Each one affords different properties; with varying degrees of security, speed, and control.
+
+By standardizing a cross-chain execution interface, we can cleanly separate the transport layer from the application layer. Cross-chain execution becomes another composable piece with which the Ethereum ecosystem can build shared infrastructure.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+This specification allows contracts on one chain to send messages to contracts on another chain. There are two key interfaces that needs to be implemented:
+
+- CrossChainRelayer
+- CrossChainExecutor
+
+The `CrossChainRelayer` lives on the origin chain. Users can share a single `CrossChainRelayer` or deploy their own.
+
+The `CrossChainExecutor` lives on the destination chain and executes relayed calls. Users can share a single `CrossChainExecutor` or deploy their own.
+
+### CrossChainRelayer
```

Comment:
Should this be in code blocks?
```suggestion
### `CrossChainRelayer`
```

Reply:
It's a title, so it doesn't really make sense to put it in code blocks.